docs/using/plugins.md

```diff
@@ -1,65 +1,88 @@
     * Each of these attributes should be tagged with units and have default values (which can be `None`)
 
 * The "type class" _may_
-  * override methods like `__init__` for custom behavior, but should be avoided if possible
-  * define its own `@property`s, class methods, etc. if needed
+  * override methods like `__init__` for custom behavior if required (this should be avoided if possible)
+  * define its own properties, class methods, etc. as needed
   * define optional attributes such as `id`
 
-## Creating custom `SMIRNOFFCollection`s
+[`ParameterType`]: openff.toolkit.typing.engines.smirnoff.ParameterType
+[entry point group]: setuptools:userguide/entry_point
+[`check_handler_compatibility`]: openff.toolkit.typing.engines.smirnoff.parameters.ParameterHandler.check_handler_compatibility
+
+## Creating a custom `SMIRNOFFCollection`
 
 There is similarly no specification for these plugins yet, but some guidelines that should be followed.
 
+A `SMIRNOFFCollection` is a subclass of [`Collection`] specific to SMIRNOFF force fields. The objectives of a collection are to:
+
+* store a system's parameters, as assigned by SMIRKS-based typing of chemical topologies
+* provide an interface for exporting these parameters to MD engines (i.e. creating an [`openmm.Force`])
+
 * The class _should_
-  * inherit from `SMIRNOFFCollection` and therefore be written in the style of a [Pydantic model](https://docs.pydantic.dev/usage/models/).
-  * define a field `is_plugin: bool = True`
-  * be available in the `openff.interchange.plugins.collections` entry point group
-  * define a field `type`, suggested to match the corresponding `ParameterHandler._TAGNAME`
-  * define a field `expression`, a string defining how it contributes to the overall potential energy
-  * define a class method `allowed_parameter_handlers` that returns an iterable of `ParameterHandler` subclasses that it can process
-  * define a class method `supported_parameters` that returns an interable of parameters that it expects to store (i.e. `smirks`, `k`, `length`, etc.)
-  * override methods of `SMIRNOFFCollection` (`store_matches`, `store_potentials`, `create`) as needed
+  * inherit from [`SMIRNOFFCollection`] and therefore be written in the style of a [Pydantic model].
+  * define a field [`is_plugin: bool = True`]
+  * be available in the `openff.interchange.plugins.collections` [entry point group]
+  * define a field [`type: str`], which we suggest match the corresponding `ParameterHandler._TAGNAME`
+  * define a field [`expression: str`], a mathematical formula defining how it contributes to the overall potential energy
+  * define a class method [`allowed_parameter_handlers`] that returns an iterable of the `ParameterHandler` subclasses that it can process
+  * define a class method [`supported_parameters`] that returns an iterable of parameters that it expects to store (i.e. `"smirks"`, `"k"`, `"length"`, etc.)
```

Comment:
Out of scope here, but I think a sore point in our infrastructure is that "parameter" can easily refer to two or more things, one of which contains the other

```
x = sage["vdW"].parameters[-1]
y = sage["vdW"].parameters[-1].sigma
```

if, using conversational English, I was trying to describe something like "grab this parameter, do this thing, ..." you'd have no clue which I was referring to. I intuitively think that `y` is more accurately described as a "parameter" but I have a hard time saying that indexing into `ParameterHandler.ParameterList` doesn't also give you a "parameter" !

I don't  help the cause on the Interchange side of things - while I find this is natural to work with:

```python3
In [15]: [*out["vdW"].potentials.values()][-1]
Out[15]: Potential(parameters={'sigma': <Quantity(0.534539231, 'angstrom')>, 'epsilon': <Quantity(1.23259997e-05, 'kilocalorie / mole')>}, map_key=None)

In [16]: [*out["vdW"].potentials.values()][-1].parameters
Out[16]:
{'sigma': 0.5345392308836691 <Unit('angstrom')>,
 'epsilon': 1.232599966667e-05 <Unit('kilocalorie / mole')>}
```
it's hard to argue that it doesn't further overload the word.

Totally out of scope here - wanted to jot my thoughts on this wart _somewhere_ - feel free to move this to an issue somewhere if you think it's feasible for us to make progress on this.

Reply:
Mmm true. This seems like its also at the root of how we have a type named `ParameterType`, which we apparently sometimes use as `Type[ParameterType]`... Maybe if we had another word that meant the `x = sage["vdW"].parameters[-1]` parameter we could rename that type to match and simplify things.